src/core/cgroup-bpf-ctx.c

```diff
@@ -1,20 +1,82 @@
 /* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "bpf-program.h"
 #include "cgroup-bpf-ctx.h"
 #include "cgroup.h"
 #include "memory-util.h"
 #include "strv.h"
 #include "string-util.h"
+#include "strv.h"
+
+int pinned_bpf_program_from_string(const char *str, enum bpf_attach_type *attach_type, char **bpffs_path) {
```

Comment:
we usually prefix return params with `ret_` to make clear they are return params...